src/pykeen/models/unimodal/boxe_kg.py

```diff
@@ -0,0 +1,196 @@
+from typing import Any, ClassVar, Mapping, Optional
+
+from torch.nn.init import normal_, uniform_, zeros_
+from ...nn.modules import Interaction
+from ...losses import NSSALoss
+import torch
+from ...models import ERModel
+from ...nn.emb import EmbeddingSpecification
+from ...constants import DEFAULT_EMBEDDING_HPO_EMBEDDING_DIM_RANGE
+from ...typing import Hint, Initializer
+
+__all__ = [
+    'BoxEKG',
+]
+
+
+SANITY_EPSILON = 10 ** -8
+
+
+def product_normalise(input_tensor):   # Geometric normalize
```

Comment:
needs type annotations and a docstring. what is the purpose of this function? can you write out the math that coresponds to this using latex in the docstring? use triple-double quotes and the sphinx documentation style

```suggestion
def product_normalise(input_tensor: torch.FloatTensor) -> torch.FloatTensor:
```